vendor/github.com/go-jose/go-jose/v3/encoding.go

```diff
@@ -183,3 +183,9 @@ func (b byteBuffer) bigInt() *big.Int {
 func (b byteBuffer) toInt() int {
 	return int(b.bigInt().Int64())
 }
+
+// base64URLDecode is implemented as defined in https://www.rfc-editor.org/rfc/rfc7515.html#appendix-C
+func base64URLDecode(value string) ([]byte, error) {
+	value = strings.TrimRight(value, "=")
+	return base64.RawURLEncoding.DecodeString(value)
+}
```

Comment:
This was added in https://github.com/go-jose/go-jose/commit/d7b900b5c5f8b9b2fec8aa1fe31380361b838bc2. I'm not sure it's correct. The linked commit cites an appendix of RFC 7515 with an example implementation, but I don't think that implementation is normative.

https://www.rfc-editor.org/rfc/rfc7515.html#section-2 has the normative definition of base64url, which does not treat the padding as optional but says "Base64 encoding using the URL- and filename-safe character set defined in [Section 5 of RFC 4648](https://www.rfc-editor.org/rfc/rfc4648#section-5) [[RFC4648](https://www.rfc-editor.org/rfc/rfc4648)], with all trailing '=' characters omitted"

Reply:
Agreed that treating the padding characters as optional is not correct -- if you're using base64url, they are not supposed to be there at all. But as the commit message says, they've seen it in the wild, and so have we -- we allow folks to send us CSRs with padding even though the ACME spec says they're not supposed to. So I'm not totally surprised they're being liberal in what they accept here.